plutus-core/plutus-ir/src/PlutusIR/Transform/CommuteFnWithConst.hs

```diff
@@ -0,0 +1,130 @@
+{-# LANGUAGE GADTs            #-}
+{-# LANGUAGE LambdaCase       #-}
+{-# LANGUAGE TypeApplications #-}
+
+module PlutusIR.Transform.CommuteFnWithConst
+  (commuteFnWithConst
+  , commuteDefaultFun)
+  where
+
+import Control.Lens (over)
+import Data.Typeable (Typeable, eqT)
+import PlutusCore.Default
+import PlutusIR.Core.Plated (termSubterms)
+import PlutusIR.Core.Type (Term (Apply, Builtin, Constant))
+
+{- | Commute such that constants are the first arguments. Consider:
```

Comment:
This doc comment attaches to `isConstant`, which I suspect isn't what you wanted. Do you want to put it as the module haddock?